abstractions/go/request_info.go

```diff
@@ -0,0 +1,94 @@
+package abstractions
+
+import (
+	"errors"
+	"reflect"
+
+	u "net/url"
+
+	s "github.com/microsoft/kiota/abstractions/go/serialization"
+)
+
+/* This type represents an abstract HTTP request. */
+type RequestInfo struct {
+	Method          HttpMethod
+	URI             u.URL
+	Headers         map[string]string
+	QueryParameters map[string]string
+	Content         []byte
+	options         map[string]MiddlewareOption
+}
+
+func (request *RequestInfo) AddMiddlewareOptions(options ...MiddlewareOption) error {
+	if options == nil {
+		return errors.New("MiddlewareOptions cannot be nil")
+	}
+	if request.options == nil {
+		request.options = make(map[string]MiddlewareOption, len(options))
+	}
+	for _, option := range options {
+		tp := reflect.TypeOf(option)
+		name := tp.Name()
+		request.options[name] = option
+	}
+	return nil
```

Comment:
Do we really want to return nil in this case?

Reply:
Marking this a global comment in other functions that involve passing by value,

Reply:
in go you return exceptions (errors) instead of throwing them. If you don't return anything but indicate in the prototype the method returns an error, the compiler will complain. This essentially means "everything went well".

Reply:
Thanks for that!